diff --git a/zombie/iso/objects/IsoTree.java b/zombie/iso/objects/IsoTree.java
index c9536ee..a1252f7 100644
--- a/zombie/iso/objects/IsoTree.java
+++ b/zombie/iso/objects/IsoTree.java
@@ -288,7 +288,7 @@ public class IsoTree extends IsoObject {
          int var80 = IsoCamera.frameState.playerIndex;
          float var81 = this.getAlpha(var80);
          float var82 = this.getTargetAlpha(var80);
-         this.setAlphaAndTarget(var80, 0.25F);
+         this.setAlphaAndTarget(var80, zombie.transparent_all.TransparentAll.XRAY_ALPHA);
          this.renderInner(var1, var2, var3, var4, true, false);
          this.setAlpha(var80, var81);
          this.setTargetAlpha(var80, var82);
