diff --git a/zombie/core/textures/MultiTextureFBO2.java b/zombie/core/textures/MultiTextureFBO2.java
index be18441..2325015 100644
--- a/zombie/core/textures/MultiTextureFBO2.java
+++ b/zombie/core/textures/MultiTextureFBO2.java
@@ -21,7 +21,7 @@ import zombie.network.GameServer;
 import zombie.network.ServerGUI;
 
 public final class MultiTextureFBO2 {
-   private final float[] zoomLevelsDefault = new float[]{2.5F, 2.25F, 2.0F, 1.75F, 1.5F, 1.25F, 1.0F, 0.75F, 0.5F, 0.25F};
+   private final float[] zoomLevelsDefault = new float[]{8.0F, 4.0F, 3.0F, 2.5F, 2.25F, 2.0F, 1.75F, 1.5F, 1.25F, 1.0F, 0.75F, 0.5F, 0.25F, 0.12F};
    private float[] zoomLevels;
    public TextureFBO current;
    public volatile TextureFBO fboRendered;
