diff --git a/zombie/iso/LightingJNI.java b/zombie/iso/LightingJNI.java
index e15a0f6..2da3857 100644
--- a/zombie/iso/LightingJNI.java
+++ b/zombie/iso/LightingJNI.java
@@ -1265,6 +1265,7 @@ public final class LightingJNI {
 
       @Override
       public int lightverts(int i) {
+         if (zombie.admin_xray.AdminXray.losOverride) return zombie.admin_xray.AdminXray.FULL_VERT_LIGHT;
          return this.cacheVertLight[i];
       }
 
@@ -1285,34 +1286,40 @@ public final class LightingJNI {
 
       @Override
       public boolean bSeen() {
+         if (zombie.admin_xray.AdminXray.losOverride) return true;
          this.update();
          return (this.vis & 1) != 0;
       }
 
       @Override
       public boolean bCanSee() {
+         if (zombie.admin_xray.AdminXray.losOverride) return true;
          this.update();
          return (this.vis & 2) != 0;
       }
 
       @Override
       public boolean bCouldSee() {
+         if (zombie.admin_xray.AdminXray.losOverride) return true;
          this.update();
          return (this.vis & 4) != 0;
       }
 
       @Override
       public float darkMulti() {
+         if (zombie.admin_xray.AdminXray.losOverride) return 1.0F;
          return this.cacheDarkMulti;
       }
 
       @Override
       public float targetDarkMulti() {
+         if (zombie.admin_xray.AdminXray.losOverride) return 1.0F;
          return this.cacheTargetDarkMulti;
       }
 
       @Override
       public ColorInfo lightInfo() {
+         if (zombie.admin_xray.AdminXray.losOverride) return zombie.admin_xray.AdminXray.WHITE;
          this.update();
          return this.lightInfo;
       }
