diff --git a/zombie/radio/ZomboidRadio.java b/zombie/radio/ZomboidRadio.java
index c273866..2f329b3 100644
--- a/zombie/radio/ZomboidRadio.java
+++ b/zombie/radio/ZomboidRadio.java
@@ -187,7 +187,7 @@ public final class ZomboidRadio {
          zombie.GameWindow.WriteString(var1.bb, (String)var3.getKey());
          var1.putInt(((Map)var3.getValue()).size());
 
-         for (Entry var5 : ((Map)var3.getValue()).entrySet()) {
+         for (Entry var5 : (java.util.Set<Entry>)((Map)var3.getValue()).entrySet()) {
             var1.putInt((Integer)var5.getKey());
             zombie.GameWindow.WriteString(var1.bb, (String)var5.getValue());
          }
@@ -774,12 +774,7 @@ public final class ZomboidRadio {
    }
 
    private String applyWeatherInterference(String var1, int var2) {
-      if (ClimateManager.getInstance().getWeatherInterference() <= 0.0F) {
-         return var1;
-      } else {
-         int var3 = (int)(ClimateManager.getInstance().getWeatherInterference() * 100.0F);
-         return this.scrambleString(var1, var3, var2 == -1);
-      }
+      return var1;
    }
 
    private String scrambleString(String var1, int var2, boolean var3) {
